### PR TITLE
Scatter op

### PR DESCRIPTION
**Pull request by scxiao**
Used by the mask_rcnn and faster_rcnn

### CODE REVIEW

**Review by kahmed10 — APPROVED** [review]

File: src/targets/gpu/device/scatter.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    auto axis_dim_size = ds.lens()[axis];
+    hip_visit_all(result, arg0, inds)([&](auto output, auto data, auto s1) {
+        auto* output_ptr     = device_cast(output.data());
+        const auto* data_ptr = device_cast(data.data());
+        gs_launch(stream, ds.elements(), 256)([=](auto i)
+                                                  __device__ { output_ptr[i] = data_ptr[i]; });
+        hip_visit_all(arg1, arg2)([&](auto indices, auto update) {
+            const auto* upd_ptr     = device_cast(update.data());
+            const auto* indices_ptr = device_cast(indices.data());
+            gs_launch(stream, inds.elements(), 256)([=](auto i) __device__ {
```
Comment (pfultz2) [inline src/targets/gpu/device/scatter.cpp:27]: Remove the `256` and use the default.

Comment (scxiao) [reply src/targets/gpu/device/scatter.cpp:26]: fixed

Comment (pfultz2) [reply src/targets/gpu/device/scatter.cpp:26]: The `256` is still there on this line. It was only fixed on the line above.

Comment (scxiao) [reply src/targets/gpu/device/scatter.cpp]: Sorry, missed one kernel. Changed this time. Thanks!

**Review by pfultz2 — APPROVED** [review]

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**